src/foundry/client-esm/canvas/tokens/ring.d.mts

```diff
@@ -1,5 +1,180 @@
+   * @defaultValue `0xFFFFFF`
+   */
+  ringColorLittleEndian: number;
+
+  /**
+   * Little endian format =\> BBGGRR
+   * @defaultValue `0xFFFFFF`
+   */
+  bkgColorLittleEndian: number;
+
+  /**
+   * @defaultValue `null`
+   */
+  defaultRingColorLittleEndian: number | null;
+
+  /**
+   * @defaultValue `null`
+   */
+  defaultBackgroundColorLittleEndian: number | null;
+
+  /**
+   * @defaultValue `0`
+   */
+  effects: number;
+
+  /**
+   * @defaultValue `1`
+   */
+  scaleCorrection: number;
+
+  /**
+   * @defaultValue `1`
+   */
+  scaleAdjustmentX: number;
+
+  /**
+   * @defaultValue `1`
+   */
+  scaleAdjustmentY: number;
+
+  /**
+   * @defaultValue `1`
+   */
+  subjectScaleAdjustment: number;
+
+  /**
+   * @defaultValue `1`
+   */
+  textureScaleAdjustment: number;
+
+  colorBand: TokenRing.RingColorBand;
+
+  /** Reference to the token that should be animated */
+  get token(): Token.ConfiguredInstance | void;
+
+  /**
+   * Configure the sprite mesh.
+   * @param mesh  - The mesh to which TokenRing functionality is configured.
+   */
+  configure(mesh?: PrimarySpriteMesh): void;
+
+  /** Clear configuration pertaining to token ring from the mesh. */
+  clear(): void;
+
+  /** Configure token ring size. */
+  configureSize(): void;
+
+  /** Configure the token ring visuals properties. */
+  configureVisuals(): void;
+
+  /**
+   * Flash the ring briefly with a certain color.
+   * @param color             - Color to flash.
+   * @param animationOptions  - Options to customize the animation. @defaultValue `{}`
+   */
+  flashColor(Color: Color, animationOptions: CanvasAnimationOptions): Promise<boolean | void>;
+
+  /**
+   * Create an easing function that spikes in the center. Ideal duration is around 1600ms.
+   * @param spikePct  - Position on [0,1] where the spike occurs. @defaultValue `0.5`
+   */
+  static createSpikeEasing(spikePct?: number): (t: number) => number;
+
+  /**
+   * Easing function that produces two peaks before returning to the original value. Ideal duration is around 500ms.
+   * @param pt  - The proportional animation timing on [0,1].
+   * @returns   - The eased animation progress on [0,1].
+   */
+  static easeTwoPeaks(pt: number): number;
+
+  /**
+   * To avoid breaking dnd5e.
+   * @deprecated since v12
+   */
+  configureMesh(): void;
+
+  /**
+   * To avoid breaking dnd5e.
+   * @deprecated since v12
+   */
+  configureNames(): void;
+}
+
+declare namespace TokenRing {
+  /** The start and end radii of the token ring color band. */
+  interface RingColorBand {
+    /** The starting normalized radius of the token ring color band. */
+    startRadius: number;
+
+    /** The ending normalized radius of the token ring color band. */
+    endRadius: number;
+  }
+
+  interface Effects
+    extends Readonly<{
+      DISABLED: 0x00;
+      ENABLED: 0x01;
+      RING_PULSE: 0x02;
+      RING_GRADIENT: 0x04;
+      BKG_WAVE: 0x08;
+      INVISIBILITY: 0x10; // or spectral pulse effect
+    }> {}
```

Comment:
This is the only one I'm not *quite* sure on, whether it should be typed as such, with a more specific brand, or just as `Readonly<string, number>`